backend/layers/persistence/persistence.py

```diff
@@ -0,0 +1,68 @@
+from typing import List
+from backend.corpora.common.entities.dataset import Dataset
+from backend.layers.common.entities import Collection, CollectionMetadata, CollectionVersion, DatasetAsset, DatasetMetadata, DatasetStatus
+
+
+class DatabaseProviderInterface:
+    def create_collection(self, collection_metadata: CollectionMetadata):
+        pass
+
+    def get_collection(self, collection_id: str) -> Collection:
```

Comment:
wondering if we'll ever need to get the collection itself.  maybe this can be `get_latest_collection_version(self, collection_id: str)`, from which one can retrieve the canonical parent Collection.

Reply:
I was thinking that some of the curation API endpoints would only want collection information and not version, so I created this method. I agree though that this layer should probably be very explicit and always return "versioned" objects.